server/api/utils/clients/async_nuclio.py

```diff
@@ -102,3 +143,88 @@ def _handle_error_response(
         self._logger.warning("Request to nuclio failed. Reason:", **log_kwargs)
 
         mlrun.errors.raise_for_status(response, error_message)
+
+    def _generate_nuclio_api_gateway_body(
+        self,
+        project_name,
+        api_gateway_name,
+        functions,
+        host,
+        path,
+        description="",
+        username=None,
+        password=None,
+        canary=None,
+    ) -> dict:
+        if not functions:
+            raise ValueError("functions should contain at least one object")
+        host = (
+            f"{api_gateway_name}-{project_name}.{self._nuclio_domain[self._nuclio_domain.find('.')+1:]}"
+            if not host
+            else host
+        )
```

Comment:
```suggestion
        host = host or f"{api_gateway_name}-{project_name}.{self._nuclio_domain[self._nuclio_domain.find('.')+1:]}"
```

A little bit shorter 😄